src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph.cs

```diff
@@ -293,6 +380,105 @@ internal ProjectDependencyGraph WithAdditionalProjectReferences(ProjectId projec
             return builder.ToImmutable();
         }
 
+        /// <summary>
+        /// Computes a new <see cref="_reverseTransitiveReferencesMap"/> for the removal of a project.
+        /// </summary>
+        private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> ComputeNewReverseTransitiveReferencesMapForRemovedProject(
+            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> existingReverseTransitiveReferencesMap,
+            ProjectId projectId)
+        {
+            var builder = existingReverseTransitiveReferencesMap.ToBuilder();
+
+            foreach (var (project, references) in existingReverseTransitiveReferencesMap)
+            {
+                if (references.Contains(projectId))
+                {
+                    // Invalidate the cache for projects that reference the removed project
+                    builder.Remove(project);
+                }
+            }
+
+            builder.Remove(projectId);
+            return builder.ToImmutable();
+        }
+
+        private ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> ComputeNewReverseTransitiveReferencesMapForRemovedProjectReference(
+            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> existingReverseTransitiveReferencesMap,
+            ProjectId projectId,
+            ProjectId referencedProjectId)
+        {
+            var builder = existingReverseTransitiveReferencesMap.ToBuilder();
+
+            // Invalidate the transitive reverse references from every project previously referenced by the original
+            // project (transitively), except for cases where the removed project reference could not impact the result.
+            foreach (var (project, references) in existingReverseTransitiveReferencesMap)
+            {
+                if (!references.Contains(referencedProjectId))
+                {
+                    // If projectId references project, it isn't through referencedProjectId so the change doesn't
+                    // impact the dependency graph.
+                    continue;
+                }
```

Comment:
I'll be honest, I'm not following this bit. Can we get an example added both ways to make it clearer?

Reply:
I updated this comment. Please let me know if the new examples are clear.